resources/indexer/directory_scanner.py

```diff
@@ -0,0 +1,63 @@
+"""Recursively scan directory paths and print all file paths."""
+
+import argparse
+import logging
+import os
+from concurrent.futures import ProcessPoolExecutor
+from time import sleep
+
+
+def process_dir(path):
+    """Print out file paths and return sub-directories."""
+    try:
+        scan = list(os.scandir(path))
```

Comment:
Is there a reason for this to be a list?  It looks like the default return value of a generator would work also.

Reply:
Good point.